src/libraries/icubmod/embObjMotionControl/eomcParser.h

```diff
@@ -56,6 +57,181 @@ namespace yarp {
 //    PidAlgo_currentInnerLoop =2
 //} PidAlgorithmType_t;
 
+class ITemperatureSensor
+{
+
+public:
+    // virtual ~ITemperatureSensor() = default; //TODO: note -> if I use smart pointer I do not need the destructor - theoretically, since it gets deleted when it goes out of scope - 
+
+    virtual void setTemperatureSensor() = 0;
+    virtual double convertTempCelsiusToRaw(const double temperature) = 0;
+
+    virtual bool convertRawToTempCelsius(double* temperature) = 0;
+};
+
+class TemperatureSensorPT100 : public ITemperatureSensor
+{
+private:
+    int _r_1;
+    int _r_2;
+    int _r_3;
+    
+    double _ptc_offset;
+    double _ptc_gradient;
+    double _pga_gain;
+
+    int _vcc;
+    double _resolution_pga;
+    double _resolution_tdb;
+
+    bool isConfigured = false;
+
+public:
+
+    virtual void setTemperatureSensor()
+    {
+        _r_1 = 4700;
+        _r_2 = 4700;
+        _r_3 = 1000;
+
+        _ptc_offset = 100;
+        _ptc_gradient = 3.851;
+        _pga_gain = 2;
+        _vcc = 5;
+        _resolution_pga = 2.048;
+        _resolution_tdb = 32767;
+
+        bool isConfigured = true;
+    }
+
+    virtual double convertTempCelsiusToRaw(const double temperature) override
+    {
+        double res = 0;
+        if (!isConfigured)
+        {
+            yError("Cannot proceed, class paramters not confgured");
+            return res;
+        }
+       	
+        double tmp = (( (_ptc_offset + _ptc_gradient * temperature) / ((double)_r_1 + (_ptc_offset + _ptc_gradient * temperature))) - (double)((double)_r_3 / ((double)_r_2 + _r_3))) * (double)_vcc;
```

Comment:
From the performance point of view, it is better if this calculation is performed at once in the creation phase. You can save the tmp value in the obj.